actions/launch_job/action.yml

```diff
@@ -0,0 +1,50 @@
+name: "Dagster Serverless Job Launch"
+description: "Pushes code locations to Dagster Cloud serverless branch deployments"
+inputs:
+  organization_id:
+    description: "Dagster Cloud organization ID"
+    required: true
+  dagster_cloud_api_token:
+    description: "Dagster Cloud API token"
+    required: true
+  location_name:
+    required: true
+    description: "The name of the deployed location to launch a job from."
+  repository_name:
+    required: true
+    description: "The name of the repository to launch a job from."
+  job_name:
+    required: true
+    description: "The name of the job to launch."
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout target repo
+      uses: actions/checkout@v3
+      with:
+        ref: ${{ github.sha }}
+
+    - name: Checkout action repo
+      uses: actions/checkout@v3
+      with:
+        repository: dagster-io/dagster-cloud-action
+        ref: ${{ env.DAGSTER_ACTION_REF }}
+        path: ./action-repo/
+      env:
+        # We need to make sure we are grabbing the right ref for both external invocation of the
+        # actions (std) as well as internal invocations of the actions (internal tests).
+        # We need to get the local action ref, but this is available in the evaluation of `env`, not
+        # during the run time evaluation or the parameter evaluation:
+        # https://github.com/orgs/community/discussions/25283
+        DAGSTER_ACTION_REF: ${{ github.repository != 'dagster-io/dagster-cloud-action' && github.action_ref || github.head_ref }}
```

Comment:
If this yml file needs to change in sync with the utils/run action, will it use the action_ref version of this yml and the PR version of utils/run?

Maybe this ref should only be set outside this file (in our internal testing yml)?